_______________
## Задание 1:

На складе лежат разные фрукты в разном количестве.

Нужно написать функцию, которая на вход принимает любое количество названий фруктов и их количество,

а возвращает общее количество фруктов на складе.

In [25]:
def warehouse_accounting(**kwargs) -> int:

    """key ожидает str, value ожидает int"""
    
    return sum(kwargs.values())

_________________

## Задание 2:

Дан список с затратами на рекламу. Но в данных есть ошибки, некоторые затраты имеют отрицательную величину. 

Удалите такие значения из списка и посчитайте суммарные затраты.

Используйте _list comprehensions._


In [26]:
expenses = [100, 125, -90, 345, 655, -1, 0, 200]

In [27]:
result = [item for item in expenses if item > 0]

sum(result)

1425

Cуммарные затраты на рекламу составляют: 1425 у.е.
____________________

## Задание 3:
Даны два списка, дата покупки и суммы покупок по датам:

In [28]:
dates = ['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', 
        '2021-09-30', '2021-12-22', '2021-11-29', '2021-12-24', '2021-11-26', 
        '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', 
        '2021-10-02', '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', 
        '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', '2021-10-31', 
        '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09']

revenue = [1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 
           651, 7037, 4274, 6275, 4988, 6930, 2971, 6592, 2004, 2822, 519, 
           3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]

###     Задание 3.1:
Найдите, какая выручка у компании в ноябре,

используйте _list comprehensions_.

In [29]:
result_nov = [revenue[i] for i in range(len(dates)) if dates[i][5:-3] == '11']

sum(result_nov)

25098

Выручка компании за ноябрь составляет: 25 098 у.е.

### Задание 3.2:
Найдите выручку компании в зависимости от месяца.

Для этого напишите функцию, которая на вход принимает список с датами и список с выручкой, 

а на выходе словарь, где ключи - это месяцы, а значения - это выручка.

Используйте аннотирование типов.

In [30]:
def sort_by_month(dates: list, revenue: list) -> dict:

    """В dates ожидается список строк 
        содержащих даты в формате 'гггг-мм-дд'.
    В revenue ожидается список целых чисел."""
    
    result: dict = {}
    for i in range(len(dates)):
        if dates[i][5:-3] not in result:
            result[dates[i][5:-3]] = revenue[i]
        else:
            result[dates[i][5:-3]] += revenue[i]
    return result

In [31]:
sort_by_month(dates, revenue)

{'09': 25647, '12': 45452, '10': 28645, '11': 25098}

Выручка компании по месяцам:

- Сентябрь: 25 647 у.е.
- Октябрь: 28 645 у.е
- Ноябрь: 25 098 у.е
- Декабрь: 45 452 у.е
______________